lightning/src/util/events.rs

```diff
@@ -67,6 +68,38 @@ pub enum PaymentPurpose {
 	SpontaneousPayment(PaymentPreimage),
 }
 
+#[derive(Clone, Debug)]
+/// Some information provided on the closure source of the channel halting.
```

Comment:
Maybe "The reason which the channel was closed. See individual variants for more details." or something. I don't think we use "halting" elsewhere in our codebase, so its somewhat confusing here.

Reply:
Yep, just lack inspiration on this comment without introducing redundancy for the reader. I'm going to take it.